core/build/src/lib.rs

```diff
@@ -66,24 +66,78 @@ pub fn sgx_builder() -> Builder {
         .derive_partialord(true)
         // Comments can cause doc tests to fail, see https://github.com/rust-lang/rust-bindgen/issues/1313
         .generate_comments(false)
-        .default_enum_style(EnumVariation::Consts)
+        .default_enum_style(EnumVariation::NewType { is_bitfield: false })
         .prepend_enum_name(false)
         .use_core()
         .ctypes_prefix("core::ffi")
         .allowlist_recursively(false)
-        .parse_callbacks(Box::new(SgxParseCallbacks))
+        .parse_callbacks(Box::new(SgxParseCallbacks::default()))
 }
 
 /// SGXParseCallbacks to be used with [bindgen::Builder::parse_callbacks]
 ///
 /// This provides a default implementation for most of the SGX libraries
-#[derive(Debug)]
-pub struct SgxParseCallbacks;
+#[derive(Debug, Default)]
+pub struct SgxParseCallbacks {
+    // types that are blocklisted from deriving Clone for.
+    //
+    // These may either already have the Clone attribute from bindgen, or they
+    // can't derive clone due to having non cloneable default types like
+    // `__IncompleteArrayField`.
+    blocklisted_clone_types: Vec<String>,
+}
+
+impl SgxParseCallbacks {
+    /// SGXParseCallbacks to be used with [bindgen::Builder::parse_callbacks]
+    ///
+    /// # Arguments
+    /// * `blocklisted_clone_types` - Types to blocklist from deriving `Clone`.
+    pub fn new<I>(blocklisted_clone_types: I) -> Self
+    where
+        I: Iterator,
+        I::Item: AsRef<str>,
+        I::Item: ToOwned,
+    {
+        let blocklisted_clone_types = blocklisted_clone_types
+            .map(|s| s.as_ref().to_owned())
+            .collect::<Vec<_>>();
+        Self {
+            blocklisted_clone_types,
+        }
+    }
+
+    // Returns the derive attributes needed to implement `Clone` on `name`, if
+    // `name` should implement clone.
+    //
+    // If `name` shouldn't derive `Clone` returns an empty vector.
+    //
+    // # Arguments
+    // * `name` - The name of the type to check for deriving `Clone`.  The name
+    //  is expected to be the _final_ name after any normalization,
+    //  [bindgen::Builder::ParceCallbacks::item_name]
+    fn maybe_derive_clone(&self, name: &str) -> Vec<String> {
+        if self.blocklisted_clone_types.iter().any(|n| *n == name) {
+            vec![]
+        } else {
+            // In order to support packed types we also need to derive `Copy`,
+            // https://github.com/rust-lang/rust/issues/82523
+            vec!["Clone".to_string(), "Copy".to_string()]
+        }
+    }
```

Comment:
pulled functions back in

Reply:
note: Had to undo the `normalize_item_name()` because of https://github.com/mobilecoinfoundation/sgx/blob/main/tstdc/sys/types/build.rs#L12  
I'll need to do some more learning on supertraits, it looks like you have to own both traits in the crate to do something like, https://riptutorial.com/rust/example/22917/inheritance-with-traits